### DISCUSSION TITLE
Custom options on assets

### DISCUSSION THREAD

**Discussion by the4thamigo-uk** [dagster-io/dagster#22786] [Ideas]
This suggestion is motivated by a dbt pipeline managed via dagster.

**Problem**

Sometimes we need to apply a `--full-refresh` to our dbt assets. Currently the user needs to go into the launchpad and add the setting to the run config. Several of our users find this tricky to get right as it requires familiarity with dagster and dbt internals.

**Aims**

Add a new feature into dagster UI to make it easy for the user to select certain predefined configuration settings when initiating a backfill run.

** Possible solutions **

Allow a way to configure _custom options_ to appear in the `Launch Runs` dialog that cause certain configuration settings to be applied to the run(s).

For example, for a dbt asset you would expect to see `Apply 'full refresh' to dbt assets` in the `Options` section, and this would be configured by the developer to create the appropriate config settings for the run.

**Comment by ion-elgreco** [comment]
This would be really nice, currently our users need to open the launchpad to change the write_mode of the IO manager, it would be great if you can link this to the buttons in the UI
